### PR TITLE
[FEATURE] Added status code in error formatter

### PR DESCRIPTION
**Pull request by lewisvoncken**

### CODE REVIEW

File: src/Error/FormattedError.php [hunk 1]
```diff
@@ -216,6 +216,7 @@ static function (SourceLocation $loc): array {
             $formattedError = self::addDebugEntries($formattedError, $exception, $debug);
         }
 
+        $formattedError['code'] = $e->getCode();
```
Comment (spawnia) [inline src/Error/FormattedError.php:219]: This is discouraged by the GraphQL specification, see http://spec.graphql.org/draft/#note-5c13b

File: src/Error/Error.php [hunk 2]
```diff
@@ -100,7 +100,7 @@ public function __construct(
         $previous = null,
         array $extensions = []
     ) {
-        parent::__construct($message, 0, $previous);
+        parent::__construct($message, $previous->getCode(), $previous);
```
Comment (spawnia) [inline src/Error/Error.php:103]: `$previous` can be `null`.

**Review by spawnia — CHANGES_REQUESTED** [review]

File: src/Error/FormattedError.php (hunk 1 above)
Comment (simPod) [reply src/Error/FormattedError.php:219]: We should not pursue this, right? So it stays protocol agnostic.

Comment (spawnia) [reply src/Error/FormattedError.php:219]: Well the `code` in this case is an internal PHP error code, not an HTTP status code. But as the spec says, if anywhere it should go under the `extensions` key.

**Comment by spawnia** [comment]
Closing because the reasons I outlined in https://github.com/webonyx/graphql-php/pull/869#discussion_r641364409

**Pull request closed without merging**